module/tirdad.c

```diff
@@ -146,208 +99,31 @@ u32 secure_tcpv6_seq_hooked(const __be32 *saddr, const __be32 *daddr,
-	    return 1;
-	}
-	ptep=_pte_direct(pmd, address);
-	if (!ptep){
-		return -1;
-	}
-	ent_val = *((unsigned long*)(ptep));
-	op_num = 8;
-	cbit = bits & op_num;
-	if (cbit){
-	    ent_val = ent_val | 2;
-	}else{
-	    ent_val = ent_val & ~((u8)2);
-	}
-	*((unsigned long*)ptep) = ent_val;
-	return 1;
-}
-
-int install_hook_on(struct target_vals *target){
-	char payload[] = "\x48\xB8\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xE0";
-	u8* payload_adr;
-	pgd_t *pgd;
-	p4d_t *p4d;
-	pud_t *pud;
-	pmd_t *pmd;
-	pte_t *ptep;
-	unsigned long ent_val;
-	struct mm_struct *mm;
-	unsigned short ps = 1 << 7;
-	u8 cbit;
-	u8 p_bits;
-
-	p_bits=0;
-
-	mm = current->mm;
-	pgd = pgd_offset(mm, target->adr);
-
-	if (pgd_none(*pgd) || unlikely(pgd_bad(*pgd))){
-		_s_out(1,"FATAL: Page tables not accessible.");
-		return -1;
-	}
-
-	ent_val = *((unsigned long*)pgd);
-	cbit = ent_val & 2;
-	if (cbit) p_bits = 1;
-	p4d = p4d_offset(pgd,target->adr);
-
-	pud = pud_offset(p4d, target->adr);
-	ent_val = *((unsigned long*)pud);
-	cbit = ent_val & 2;
-	if (cbit) p_bits = p_bits | 2;
-
-	if (!!( ps & ent_val ) == 1){
-	    goto install;
-	}
-
-	pmd = pmd_offset(pud, target->adr);
-	VM_BUG_ON(pmd_trans_huge(*pmd));
-	ent_val = *((unsigned long*)pmd);
-	cbit = ent_val & 2;
-	if (cbit) p_bits = p_bits | 4;
-
-	if (!!( ps & ent_val ) == 1){
-	    goto install;
-	}
-
-	ptep=_pte_direct(pmd, target->adr);
-
-	if (!ptep){
-		_s_out(1,"FATAL: Page table entry not accessible.");
-		return -1;
-	}
-
-	ent_val = *((unsigned long*)(ptep));
-	cbit = ent_val & 2;
-	if (cbit) p_bits = p_bits | 8;
-
-install:
-
-	store_p_bits(target->adr,0x0F);
-
-	payload_adr = (u8*) target->adr;
-	memcpy(target->backup_bytes,(void*)target->adr,FUSION_SIZE);
-	memcpy((void*)target->adr,payload,FUSION_SIZE);
-	*((unsigned long*)&payload_adr[2]) = target->hook_adr;
-
-	/*
-	 * Revert entries to original values.
-	*/
-	store_p_bits(target->adr,p_bits);
-
-	target->p_bits=p_bits;
-
-	return 0;
-}
-
-void recover_one(struct target_vals *target){
-	store_p_bits(target->adr,0x0F);
-	memcpy((void*)target->adr,target->backup_bytes,FUSION_SIZE);
-	store_p_bits(target->adr,target->p_bits);
-}
-
-#define SYMBOL_LOOKUP(s) (((u64 (*)(const char *))(kasln_adr))(s))
-#define HANDLER(t,l,ret,...) t l ## h_hk(struct kprobe * kp, struct pt_regs * r\
-					__VA_OPT__(,) __VA_ARGS__){\
-						ret;\
-					}
-
-HANDLER(int,pre,return 0)
-HANDLER(void,post,return,unsigned long flags)
-
-int get_kasln_adr(void){
-	struct kprobe h_kprobe;
-	int r;
-
-	memset(&h_kprobe, 0, sizeof(h_kprobe));
-	h_kprobe.pre_handler = preh_hk;
-	h_kprobe.post_handler = posth_hk;
-	h_kprobe.symbol_name = "kallsyms_lookup_name";
-	r = register_kprobe(&h_kprobe);
-	if (!r){
-		kasln_adr=(u64)h_kprobe.addr;
-	}
-	unregister_kprobe(&h_kprobe);
+static struct klp_func funcs[] = {
+ 	{
+ 		.old_name = "secure_tcp_seq",
+		.new_func = secure_tcp_seq_hooked,
+	},
+	{
+		.old_name = "secure_tcpv6_seq",
+		.new_func = secure_tcp_seq_hooked,
```

Comment:
This should be `secure_tcpv6_seq_hooked`

Reply:
ugh, you're right, good typo catch. Sorry about that.

Reply:
Force-pushed a fix.